code/crates/sync/src/handle.rs

```diff
@@ -280,67 +236,43 @@ pub async fn on_value_response<Ctx>(
 where
     Ctx: Context,
 {
-    debug!(%response.height, %request_id, %peer_id, "Received response");
+    let start = response.start_height;
+    debug!(start_height = %start, peer = %peer_id, "Received batch response with {} values", response.values.len());
 
-    state.remove_pending_value_request_by_height(&response.height);
+    // Remove pending requests for the values received
+    if !response.values.is_empty() {
+        let last_value_height = start.increment_by(response.values.len() as u64 - 1);
+        state.remove_pending_value_request_by_height_range(&(start..=last_value_height));
```

Comment:
We probably should not remove all range here. Instead remove heights when decided. The problem with this is that when we start a height, in `on_started_height` we will request again. I think you will see something like this:
- start height=1, request 1..100
- height=1 is decided, the range is removed
- start height=2:
  - consensus will have the 2..100 synced values buffered (after Romain's fix)
  - sync runs `on_started_height(2)` and will request 2..101
so we request again, consensus will end up with duplicate sync values